lib/datatype/index.js

```diff
@@ -72,25 +100,44 @@ export class DataType {
     if (decode) {
       this.#decode = decode
     }
+
+    const secretKey = this.#keyPair.secretKey
+    const publicKey = this.#keyPair.publicKey
+
+    this.#writer = this.#corestore.get({
```

Comment:
this is actually the same writer hypercore defined in datastore, so it may be that we can just pass that core in this constructor. the only difference really is just that we wouldn't be repeating this `corestore.get` boilerplate.

Reply:
I think this would be a good change to make, because if someone were to change the sign or verify functions in one place but not another I think that would create problems?